### DISCUSSION TITLE
Bulk Contact Assignments/Updates

### DISCUSSION THREAD

**Discussion by jdavidson2021** [netbox-community/netbox#8413] [General]
Is there currently a way to bulk assign a contact to multiple devices at once?  We have added a few contacts, contact groups, and contact roles, but assigning a contact to a device one at a time in the GUI is not really feasible with 1000s of devices.  It would be nice to be able to do a bulk edit on a bunch of existing devices and assign the contact(s).  It would also be nice to be able to include the contact(s) details when you do a bulk import of devices.  Am I just missing something or should I submit a feature request?

**Comment by stormlogix** [comment]
Suggestion: Add the ability to assign contacts to tags. This would solve all mentioned issues. Plus you wouldn't need to add yet another field "contacts" for all objects.

**Comment by candlerb** [comment]
Could you assign a Tenant to each device, and contacts to the Tenant?

Bulk-updating Devices to set Tenant is available in the GUI.

**Comment by tyler-8** [comment]
I was about to open an FR on this before I found this discussion and issue #8419 which is similar.

Here's a use case that may help visualize the flow:

I have provider `Initech` that has 100 circuits assigned to Sites in the United States. I have another 50 `Initech` Circuits assigned to Sites in Canada.

I have a Contact for `Initech`'s account manager role, and that Contact should be assigned to all 150 Circuits. However, the 100 US Circuits have 2 Contacts with the "NOC" Role, and the 50 Circuits in Canada have a different 2 Contacts for "NOC".

Rather than adding those two "sets" of Contacts individually to each Circuit, I could create 2 Contact "groups/templates/buckets/etc" 

Bucket 1
- Initech Account Manager A
- Initech US NOC1
- Initech US NOC2

Bucket 2
- Initech Account Manager A
- Initech Canada NOC1
- Initech Canada NOC2

then I could more easily apply Bucket 1 to 100 Circuits, and Bucket to 50 circuits. This is just one of many similar workflows that I can see this filling a void for. The terminology for "Contract Groups" is already being used so I'm not sure how to deal with that, maybe "Contact Templates" would be viable?
